src/core/createEventManager.js

```diff
@@ -35,6 +35,11 @@ export default ({ createEvent, optIn, lifecycle, network, config }) => {
         })
         .then(() => {
           applyUserProvidedData(event);
+          try {
+            event.applyCallback(onBeforeEventSend);
+          } catch (e) {
+            logger.warn(e);
```

Comment:
I think `logger.error` would be more appropriate since the callback failed and therefore may be resulting in erroneous data.